datafusion/expr/src/built_in_function.rs

```diff
@@ -737,8 +732,7 @@ impl BuiltinScalarFunction {
                 Utf8 => List(Arc::new(Field::new("item", Utf8, true))),
                 Null => Null,
                 _ => {
-                    // this error is internal as `data_types` should have captured this.
-                    return internal_err!(
+                    return plan_err!(
```

Comment:
it is true that this is currently caught by higher up type checks, but I think it is better to throw a plan error here to tell users about the problem if something changes rather than throw a confusing internal error

Also given that this code is often copy/pasted it probably should default to returning a nice error message rather than an opaque one